packages/skygear-core/src/client.ts

```diff
@@ -82,6 +83,12 @@ export abstract class BaseAPIClient {
     if (this.userAgent !== undefined) {
       headers["user-agent"] = this.userAgent;
     }
+    if (this.getExtraSessionInfo) {
+      const extraSessionInfo = await this.getExtraSessionInfo();
+      if (extraSessionInfo) {
+        headers["x-skygear-extra-info"] = JSON.stringify(extraSessionInfo);
```

Comment:
Do we need to encode the json string with base64 or something else?

Reply:
JSON string is safe to be used as HTTP header value.

Reply:
I guess it is true only when string literal is in unicode code escape.

Reply:
It seems non-ASCII encoded characters in header value is undefined. I'll change to encode the JSON in base64.